hyperspy/_signals/signal2d.py

```diff
@@ -778,6 +786,126 @@ def add_ramp(self, ramp_x, ramp_y, offset=0):
+        current_index : bool
+            if True, the computation will be performed for the current index.
+        %s
+        %s
+        %s
+        %s
+
+        **kwargs : dict
+            Keywords parameters associated with above methods, see the
+            documentation of each method for more details.
+
+        Notes
+        -----
+        As a convenience, the 'local_max' method accepts the 'distance' and
+        'threshold' argument, which will be map to the 'min_distance' and
+        'threshold_abs' of the :py:func:`skimage.feature.peak_local_max`
+        function.
+
+        Returns
+        -------
+        peaks : :py:class:`~hyperspy.signal.BaseSignal` or numpy.ndarray if current_index=True
+            Array of shape `_navigation_shape_in_array` in which each cell
+            contains an array with dimensions (npeaks, 2) that contains
+            the x, y pixel coordinates of peaks found in each image.
+        """
+        method_dict = {
+            'local_max': peak_local_max,
+            'max': find_peaks_max,
+            'minmax': find_peaks_minmax,
+            'zaefferer': find_peaks_zaefferer,
+            'stat': find_peaks_stat,
+            'laplacian_of_gaussian':  find_peaks_log,
+            'difference_of_gaussian': find_peaks_dog,
+            'template_matching' : find_peaks_xc,
+        }
+        # As a convenience, we map 'distance' to 'min_distance' and
+        # 'threshold' to 'threshold_abs' when using the 'local_max' method to
+        # match with the arguments of skimage.feature.peak_local_max.
+        if method == 'local_max':
+            if 'distance' in kwargs.keys():
+                kwargs['min_distance'] = kwargs.pop('distance')
+            if 'threshold' in kwargs.keys():
+                kwargs['threshold_abs'] = kwargs.pop('threshold')
+        if method in method_dict.keys():
+            method_func = method_dict[method]
+        else:
+            raise NotImplementedError(f"The method `{method}` is not "
+                                      "implemented. See documentation for "
+                                      "available implementations.")
+        if interactive:
+            # Create a peaks signal with the same navigation shape as a
+            # placeholder for the output
+            axes_dict = self.axes_manager._get_axes_dicts(
+                self.axes_manager.navigation_axes)
+            peaks = BaseSignal(np.empty(self.axes_manager.navigation_shape),
+                               axes=axes_dict)
+            pf2D = PeaksFinder2D(self, method=method, peaks=peaks, **kwargs)
+            pf2D.gui(display=display, toolkit=toolkit)
+        elif current_index:
+            peaks = method_func(self.__call__(), **kwargs)
+        else:
+            peaks = self.map(method_func, show_progressbar=show_progressbar,
+                             parallel=parallel, inplace=False,
+                             ragged=True, **kwargs)
+
+        return peaks
```

Comment:
@dnjohnstone why not call the sort here to be certain it's the last thing that runs before it returns?

Reply:
Just in case, for example, there's anything amiss with `.map()` and multiple threads...though it fails for `parallel=False` too I guess.